src/d3/hapticplot.d3.spec.ts

```diff
@@ -1,89 +1,107 @@
   });
 
   it('places points for each datum in a three datum array', () => {
     hapticplot.init(scene, [0, 10, 20]);
     const expectedPosArray = [new Vector3(0, 0, -1), new Vector3(0.1, 10, -1), new Vector3(0.2, 20, -1)];
-    const result = getPosition(scene, shape);
+    const result = getPositions(scene, shape);
     expect(result).toEqual(expectedPosArray);
   });
 
   it('places points for each datum and sets the correct color property on the resulting shape entities', () => {
     hapticplot.init(scene, [10, 20, 30]);
     const expectedColorArray = ['green', 'green' , 'green'];
-    const result = getColor(scene, shape);
+    const result = getColors(scene, shape);
     expect(result).toEqual(expectedColorArray);
   });
 
   it('places points for each datum and sets the correct size property on the resulting shape entities', () => {
     hapticplot.init(scene, [10, 20, 30]);
     const expectedSizeArray = ['0.05', '0.05', '0.05'];
-    const result = getRadius(scene, shape);
+    const result = getRadii(scene, shape);
     expect(result).toEqual(expectedSizeArray);
   });
 
   it('places points for each datum and sets the correct hoverable property on the resulting shape entities', () => {
     hapticplot.init(scene, [10, 20, 30]);
-    const expectedSizeArray = ['hoverable', 'hoverable', 'hoverable'];
+    const expectedAttrArray = ['hoverable', 'hoverable', 'hoverable'];
     const result = getHoverable(scene, shape);
-    expect(result).toEqual(expectedSizeArray);
+    expect(result).toEqual(expectedAttrArray);
   });
 
   it('places points for each datum and sets the correct color property on the resulting shape entities after a hover event', () => {
     hapticplot.init(scene, [10, 20, 30]);
-    const expectedSizeArray = ['red', 'red', 'red'];
+    const expectedAttrArray = ['red', 'red', 'red'];
     const result = getHoveredColor(scene, shape);
-    expect(result).toEqual(expectedSizeArray);
+    expect(result).toEqual(expectedAttrArray);
+  });
+
+  it('initilizes the controller entitys and attaches the haptic component', () => {
+    const expectedRes = true;
+    const result = getHaptics(controller);
+    expect(result).toEqual(expectedRes);
   });
 });
 
 // Helper Functions
 
 // Returns an array of actual position vectors
-function getPosition(scene: HTMLElement, shape: string): Vector3[]{
+function getPositions(scene: HTMLElement, shape: string): Vector3[]{
   const attrArray: Vector3[] = [];
-  Array.from(scene.querySelectorAll(shape)).forEach((child) => {
+  const shapes = scene.querySelectorAll(shape);
+  for (const point of shapes){
     attrArray.push(new Vector3(
-      (child as Entity).object3D.position.x,
-      (child as Entity).object3D.position.y,
-      (child as Entity).object3D.position.z));
-  });
+      (point as Entity).object3D.position.x,
+      (point as Entity).object3D.position.y,
+      (point as Entity).object3D.position.z));
+  }
   return attrArray;
```

Comment:
I think you can turn this whole thing into a map:

```typescript
return scene.querySelectorAll(shape)
  .map((child: Entity) => 
    new Vector3(
      point.object3D.position.x,
      point.object3D.position.y,
      point.object3D.position.z));
```
I'm not 100% on if the type will work in the function signature, but if it doesn't, you can just use the cast like you have been